File: app/src/main/java/com/ne1c/gitteroid/ui/views/ChatView.java
package com.ne1c.gitteroid.ui.views;

import android.support.annotation.StringRes;

import com.ne1c.gitteroid.models.view.MessageViewModel;

import java.util.ArrayList;

public interface ChatView{
    void showMessages(ArrayList<MessageViewModel> messages);

    void showError(@StringRes int resId);

    void showUpdateMessage(MessageViewModel message);

    void successReadMessages(int first, int last, String roomId, int i);

    void showLoadBeforeIdMessages(ArrayList<MessageViewModel> messages);

    void deliveredMessage(MessageViewModel message);

    void errorDeliveredMessage();

    void showTopProgressBar();

    void hideTopProgressBar();

    void showListProgressBar();

    void hideListProgress();

    void joinToRoom();
}

File: app/src/main/kotlin/com/ne1c/gitteroid/ui/views/ChatView.kt
package com.ne1c.gitteroid.ui.views

import android.support.annotation.StringRes

import com.ne1c.gitteroid.models.view.MessageViewModel

import java.util.ArrayList

interface ChatView {
    fun showMessages(messages: ArrayList<MessageViewModel>)

    fun showError(@StringRes resId: Int)

    fun showUpdateMessage(message: MessageViewModel)

    fun successReadMessages(first: Int, last: Int, roomId: String, i: Int)

    fun showLoadBeforeIdMessages(messages: ArrayList<MessageViewModel>)

    fun deliveredMessage(message: MessageViewModel)

    fun errorDeliveredMessage()

    fun showTopProgressBar()

    fun hideTopProgressBar()

    fun showListProgressBar()

    fun hideListProgress()

    fun joinToRoom()
}
